### PR TITLE
ENH: add johnsonsl distribution to scipy.stats

### PR DESCRIPTION
**Pull request by mikeWShef**
also added johnsonsb to tests

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -3528,6 +3530,47 @@ def _ppf(self, q, a, b):
...
+    %(before_notes)s
+
+    See Also
+    --------
+    johnsonsb
+    johnsonsu
+
+    Notes
+    -----
+    The probability density funcition for a `johnsonsl` is::
```
Comment (chrisb83) [inline scipy/stats/_continuous_distns.py]: please align the docstring to the other distributions: use mathmode, specify range of a and b

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -3528,6 +3530,47 @@ def _ppf(self, q, a, b):
...
+        johnsonsl.pdf(x, a, b) = b/x* phi(a + b * ln (x))
+    for x>0 where ``phi`` is the normal pdf.
+
+    %(after_notes)s
+
+    .. versionadded:: 1.3.0
+
+    %(example)s
+    """
+    def _argcheck(self, a, b):
```
Comment (chrisb83) [inline scipy/stats/_continuous_distns.py]: why is this needed?

File: scipy/stats/_continuous_distns.py [hunk 3]
```diff
@@ -3528,6 +3530,47 @@ def _ppf(self, q, a, b):
...
+        return a==a
+
+    def _pdf(self, x, a, b):
+        #negatives = x<=0
```
Comment (chrisb83) [inline scipy/stats/_continuous_distns.py]: if you want to define the Distribution on [0, inf), set `a = 0` when you define `johnsonsl = johnsonsl_gen(a=0, name='johnsonsl')`

File: scipy/stats/_continuous_distns.py [hunk 4]
```diff
@@ -3490,6 +3491,7 @@ class johnsonsu_gen(rv_continuous):
     See Also
     --------
     johnsonsb
+    johnsonsl
```
Comment (chrisb83) [inline scipy/stats/_continuous_distns.py]: should be johnsonsb, johnsonsl

**Review by chrisb83 — CHANGES_REQUESTED** [review]
Thanks for creating this PR. Can you give a reference for this distribution?

are other methods known in closed form like `sf`?

**Comment by josef-pkt** [comment]
you need to check the parameterization
It looks overparameterized to me, `a` is redundant with `scale`, AFAICS

johnsonsl.pdf(x, a, b) = b/x* phi(a + b * ln (x))

johnsonsl.pdf(x, a, b, scale) = b/x* phi(a + b * ln (x / scale))     (maybe something missing)

File: scipy/stats/_continuous_distns.py (hunk 3 above)
Comment (mikeWShef) [reply scipy/stats/_continuous_distns.py]: will do thanks

File: scipy/stats/_continuous_distns.py (hunk 2 above)
Comment (mikeWShef) [reply scipy/stats/_continuous_distns.py]: it needs changing to b<0

**Comment by mikeWShef** [comment]
I think josef-pkt is right about the over parameterisation, I've re read the reference and it seems there are only 3 independent parameters for this member of the johnson family of distributions. lognormal seems to be identical with some complicated relation between the parameters. I guess that means this should just be abandoned? if it would still be useful I'm happy to do the rest of the work.

**Comment by mikeWShef** [comment]
Thats how they relate: 

sq=np.log(b/scale)*b**2

alpha=np.abs(np.exp(-1*lambertw(2*sq)/2))

lin=np.log(b/scale)*(2*a*b-2*b**2*np.log(scale))

scale_log=np.exp((lin/sq)/(-2))

shape_factor_log=1/b

loc_log=loc

**Comment by mdhaber** [comment]
@mikeWShef what did you decide on this? Just being overparameterized in its current form isn't a dealbreaker - can't you just remove the `a` parameter and use the `scale` parameter that's part of every `rv_continuous`? You can make a note in the documentation about the relationship between the different parameterizations.

Or do you mean this is equivalent to another distribution that already exists in scipy.stats?

**Comment by mikeWShef** [comment]
I meant it's equvalent to the log normal distribution, I wrote the below in my project just to have the names obvious, but any distribution you can make with this you can also make with the log normal distribution, as far as I remember using the conversion in the previous comment:
`class johnsonsl_gen(scipy.stats.rv_continuous):
    """A Johnson SL continuous random variable.
    %(before_notes)s
    See Also
    --------
    johnsonsu
    johnsonsb
    Notes
    -----
    The probability density function for `johnsonsl` is::
        johnsonsl.pdf(x, a, b) = b * phi(a + b * log(x))
    for ``0 < x < 1`` and ``a, b > 0``, and ``phi`` is the normal pdf.
    %(example)s
    """
    def _argcheck(self, a, b):  # a is gamma b is delta, I didn't really know what to do here
        return a == a

    def _pdf(self, x, a, b):
        trm = scipy.stats.norm.pdf(a+b*log(x))
        return x*trm

    def _cdf(self, x, a, b):
        return scipy.stats.norm.cdf(a+b*log(x))

    def _ppf(self, q, a, b):
        return exp((scipy.stats.norm.ppf(q)-a)/b)


johnsonsl = johnsonsl_gen(a=0.0, b=1.0, name='johnsonsl')`

**Comment by mdhaber** [comment]
OK, I've found a few places that confirm that Johnson SL is the lognormal distribution, so let's close this, as we avoid re-parameterizations of existing distributions. 
Would you be interested in contributing another distribution? There was is a stalled PR for skewed Cauchy (gh-7702) you could take over (or start over).
